contributors/design-proposals/storage/pv-to-rbd-mapping.md

```diff
@@ -0,0 +1,128 @@
+  belong to we cannot hand those customers their data.
+- Create PV with the given RBD name and pre-bind it to the desired PVC so that customer
+  can get its data back.
+
+## Non Goals
+This proposal doesnt attempt to undermine the importance of etcd backups to restore
+data in catastrophic situations. This is one additional line of defense in case our
+backups are not working.
+
+## Motivation
+
+We recently had an etcd data loss which resulted in loss of this rbd to pv mapping 
+and there was no way to restore customer data. This proposal aims to store pvc name 
+as metadata in the RBD image so that in catastrophic scenarios, the mapping can be 
+restored by just looking at the RBD's.
+
+## Current Implementation
+
+```go
+func (r *rbdVolumeProvisioner) Provision() (*v1.PersistentVolume, error) {
+...
+
+ // create random image name
+ image := fmt.Sprintf("kubernetes-dynamic-pvc-%s", uuid.NewUUID())
+ r.rbdMounter.Image = image
+```
+## Finalized Proposal
+Use `rbd image-meta set` command to store additional metadata in the RBD image about the PVC which owns
+the RBD image. 
+
+`rbd image-meta set  --pool hdd kubernetes-dynamic-pvc-fabd715f-0d24-11e8-91fa-1418774b3e9d pvcname <pvcname>`
+`rbd image-meta set  --pool hdd kubernetes-dynamic-pvc-fabd715f-0d24-11e8-91fa-1418774b3e9d pvcnamespace <pvcnamespace>`
+
+### Pros
+- Simple to implement
+- Does not cause regression in RBD image names, which remains same as earlier.
+- The metada information is not immediately visible to RBD admins
+
+### Cons
+- NA
+
+Since this Proposal does not change the RBD image name and is able to store additional metadata about 
+the PVC to which it belongs, this is preferred over other two proposals. Also it does a better job 
+of hiding the PVC name in the metadata rather than making it more obvious in the RBD image name. The
+metadata can only be seen by admins with appropriate permissions to run the rbd image-meta command. In
+addition, this Proposal , doesnt impose any limitations on the length of metadata that can be stored 
+and hence can accommodate any pvc names and namespaces which are stored as arbitrary key value pairs.
+It also leaves room for storing any other metadata about the PVC.
+
+
+### Upgrade/Downgrade Behavior
+
+#### Upgrading from a K8s version without this metadata to a version with this metadata
+The metadata for image is populated on CreateImage. After an upgrade, existing RBD Images will not have that
+metadata set. When the next AttachDisk happens, we can check if the metadata is not set, set it. Cluster
+administrators could also run a one time script to set this manually. For all newly created RBD images,
+the rbd image metadata will be set properly.
+
+#### Downgrade from a K8s version with this metadata to a version without this metadata
+After a downgrade, all existing RBD images will have the metadata set. New RBD images created after the 
+downgrade will not have this metadata. 
```

Comment:
@saad-ali @rootfs could you review this again ? My proposal is to populate the meta on CreateImage and re-verify in AttachDisk, so that in the upgrade case, all older RBD images also get the image meta . If you have a better idea, please propose.